### PR TITLE
Add Well support to wells sdk

### PR DESCRIPTION
**Pull request by 0xalecks**

### CODE REVIEW

File: projects/sdk-wells/src/lib/Well.ts [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+   * **Well functions** define a relationship between the reserves of the
+   * tokens in the Well and the number of LP tokens.
+   *
+   */
+  async getWellFunction(): Promise<WellFunction> {
+    if (!this.wellFunction) {
+      await this.getWell();
+    }
+
+    return this.wellFunction!;
```
Comment (marshall-bean) [inline projects/sdk-wells/src/lib/Well.ts:110]: I'd expect this to call the well function contract and get the name (presumably e.g. "Constant Product" or whatever).

File: projects/sdk-wells/src/lib/WellFunction.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export class WellFunction {
+  constructor(public readonly address: string, public readonly data: string) {}
```
Comment (marshall-bean) [inline projects/sdk-wells/src/lib/WellFunction.ts:2]: The IWellFunction also has a `name` property, can we surface it here?

**Review by marshall-bean — CHANGES_REQUESTED** [review]

File: projects/sdk-core/src/lib/TokenValue.ts [hunk 3]
```diff
@@ -222,4 +222,32 @@ export class TokenValue {
...
+   * Calculates value after substracting slippage.
+   *
+   * For ex, a value of 100, with slippage 3 would return 97
+   *
+   * @param slippage The percent to remove from the value. Slippage should be
+   * a human readable percentage; 3 = 3%, 25=25%, .1 = 0.1%
+   *
+   * @return The original value minus slippage
+   */
+  subSlippage(slippage: number) {
```
[the remark this answers is not in the document]
Comment (0xalecks) [reply projects/sdk-core/src/lib/TokenValue.ts:236]: TokenValue already has a .sub() method, so follows that pattern

File: projects/sdk-wells/src/lib/Well.ts [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+   * - getWell()
+   * - getLPToken()
+   * - getTokens()
+   * - getWellFunction()
+   * - getPumps()
+   * - getAuger()
+   * - getReserves()
+   * */
+  async loadWell() {
+    await Promise.allSettled([this.getName(), this.getLPToken(), this.getWell(), this.getReserves()]);
```
[the remark this answers is not in the document]
Comment (0xalecks) [reply projects/sdk-wells/src/lib/Well.ts:61]: good callout, will change to promise.all and let it fail loudly.

File: projects/sdk/src/classes/TokenValue.test.ts [hunk 5]
```diff
@@ -1,340 +0,0 @@
-import { expect } from "@jest/globals";
```
[the remark this answers is not in the document]
Comment (0xalecks) [reply projects/sdk/src/classes/TokenValue.test.ts:1]: yup, next pr when we remove TokenValue from SDK

File: projects/sdk-wells/src/lib/WellFunction.ts (hunk 2 above)
Comment (0xalecks) [reply projects/sdk-wells/src/lib/WellFunction.ts:2]: Was going to do a separate PR for these (well function, pump, auger, etc..)

**Review by marshall-bean — APPROVED** [review]

**Pull request merged**